#### Question 1

The key factor that determines whether an idea or statement can be examined and tested statistically is measurability, which refers to wheather the data can be collected, analyzed, and compared in the form of quantifiable variables which obeys the priciple of statistical analysis. There are few key criterias that a good null hypothesis should satisfy, for example specified with a clear condition, testable when proving it's wheather true or false, neutrality and universality. Null hypothesis is a default assumption which often states that the result is no effect, nor difference or no relationship with the variables for testing. An alternative hypothesis is everything else differet from the null hypothesis, which usually states that the variables being studied does have an effect on the final result. It stands for a possibility that the null hypothesis is not correct. 

https://chatgpt.com/share/66fed021-434c-800c-bf4d-98e4d3db77c6

Statistical examination requires ideas to be measurable, meaning they must involve quantifiable variables that can be objectively collected and analyzed. Without clear metrics, it becomes difficult to apply statistical methods.

A good null hypothesis is defined by a few key criteria:
1. It should be specific, clearly stating a testable condition.
2. It must be testable, allowing for data collection and statistical analysis.
3. It should be neutral, often framed as a "no effect" or "no difference" assumption.
4. It must be broad enough to encompass the scenario but focused enough to be statistically assessed.

The null hypothesis (H₀) assumes no effect or relationship between variables, serving as the default assumption. The alternative hypothesis (H₁ or Ha), by contrast, suggests that there is a significant effect or difference. Hypothesis testing evaluates whether there is sufficient evidence to reject the null hypothesis in favor of the alternative.

#### Question 2

The statement shown at the end of the vedio emphasizes the difference between sample statistics and population parameters. \
𝑥𝑖 stands for the individual data points introduced in the specific sample, where i can be  ranged from 1 to n (n is the number of data points being observed), when it points to a specific data point.\
𝑥¯ stands for the sample mean, which is an average of the 𝑥𝑖's. It is an 'statistic' which estimates the popution mean based on the specific sample being analyzed.\
𝜇 stands for the population mean, which is a 'parameter' that represenst the true mean(average) of the entire population. The population mean is usually hard to calculate since the entire population is too large when analyzing, therefore, we introduce the sample means to provide an estimation on it. \
𝜇0 stands for the hypothesized mean, which is an assumption for the value of the true population mean under the null hypothesis. The hypothesized mean is usually set to the baseline value when comparing to the sample mean 𝑥¯. During a hypothesis test, one thing we want to prove is the rejection of hypothesis '𝜇 = 𝜇0'.

https://chatgpt.com/share/66fed6e9-0fd8-800c-ab16-5a227177c665

This session focused on clarifying key statistical concepts related to hypothesis testing, specifically the distinction between observed sample values (\( x_i \)), the sample mean (\( \bar{x} \)), the population mean (\( \mu \)), and the hypothesized population mean under the null hypothesis (\( \mu_0 \)). 

- **Observed sample values (\( x_i \))** are individual data points collected from a sample.
- **The sample mean (\( \bar{x} \))** is the average of those values and serves as an estimate of the population mean, though it varies between samples.
- **The population mean (\( \mu \))** is a fixed but unknown value that represents the true average of the population.
- **The hypothesized mean (\( \mu_0 \))** is the value assumed under the null hypothesis, representing a baseline for comparison in the hypothesis test.

The distinction between these concepts highlights the goal of hypothesis testing: using sample data to draw conclusions about the population mean.

#### Question 3

In the previous questions, the null hypothesis is discribed as a 'baseline' in a hypothsis test, which is an assumption of the variable being observed does not affect the final result. A p-value is s statistical measure which helps to illustrate how the observed data is extreme from the actual data. When calculating a p=value, we assume the null hypothesis is true, this helps to evaluate how the observed data is different from the assumption. If the p-value is small, it indicates the extreme data would be unusual from the null hypothesis, and the null hypothesis is rejected; if the p-value is large, then the observation satisfies the null hypothesis, and the null hypothesis is proved since we failed to reject it.

https://chatgpt.com/share/66fedad8-2fc0-800c-99a7-b23227c2978b

In this session, two key concepts in hypothesis testing were covered:

1. **Imagining a World Where the Null Hypothesis is True:** When calculating a p-value, the assumption is made that the null hypothesis is true to evaluate how likely the observed data would be under this assumption. This helps in deciding whether to reject or fail to reject the null hypothesis based on the extremity of the data.

2. **What a p-value is:** A p-value represents the probability of obtaining a result as extreme, or more extreme, than the observed data, assuming the null hypothesis is true. A small p-value suggests evidence against the null hypothesis, while a large p-value indicates that the data is not unusual under the null hypothesis. It does not measure the probability that the null hypothesis is true or false, nor does it quantify the size or importance of an effect.

#### Question 4

As the p-value generated from a hypothesis test is smaller, it indicates that the observation data is extremely  different or unusual from the assumption under the situation when the null hypothesis is true(𝜇0 usually assumed to be no effect on the result), which rejects the null hypothesis to be a valid statement or explaination for the data. This is because, when the null hypothesis is true, there's a known dirstribution for the statistic being tested, which discribes how the data point will spread under the 'no effect' situation. During a hypothesis test, we build a test statistic from the observed sample data which summarized the how far the data should spread. After comparing the observed data and expected data under the null hypothsis, p-value is calculated to indicate the probability of the observed values being extreme from the test values. For conclusion, if the null hypothesis is true, the chance for observations to have a extreme value would be small. 



https://chatgpt.com/share/66fedad8-2fc0-800c-99a7-b23227c2978b

The observed test statistic is compared to the sampling distribution of the test statistic under the assumption that the null hypothesis is true. If the observed value falls near the center of the distribution, it suggests that the data aligns with the null hypothesis. However, if the observed statistic falls in the tails (indicating an extreme value), it suggests the result is unusual under the null hypothesis. This comparison helps determine the p-value, with more extreme test statistics leading to smaller p-values and stronger evidence against the null hypothesis.

#### Question 5

Hypotheses:\
Null hypothesis: There in 'no preference' for tilt right or left, according to the "50/50 coin-flipping" model, the probability of tilt right should be 50%. Therefore, null hypothesis set p=0.5.\
Alternative hypothesis: The probability of tilt right is greater or smaller than 50%.\

Observation:
80 of 124 couples tilt right: 80/124 = 64.5%.\
The proportion of tilt right is 64.5%

Simulation:
According to the 'coin-flip' model, if we repeat the similation for many times, the probability for 80 out of 124 couples to tilt right is 0.1%, which indicates the p-value is 0.001.

Conclusion:
since p = 0.001, there exist a very strong evidence against the null hypothesis. Therefore, we succesfully reject the null hypothesis. 


https://chatgpt.com/share/66fee8a9-9020-800c-94aa-c3f741d7e98a

In this session, a simulation was conducted to estimate a p-value for a study on head-tilting preferences among kissing couples. The null hypothesis assumed no preference for tilting direction, with a 50/50 chance of tilting left or right. The observed data showed that 80 out of 124 couples tilted their heads to the right, a proportion of 64.5%.

A binomial model representing the 50/50 probability assumption was used, and 10,000 trials were run. The resulting p-value was 0.001, indicating that the likelihood of observing 80 or more right tilts, under the assumption of no preference, is just 0.1%.

This low p-value suggests strong evidence against the null hypothesis, implying a possible preference for tilting to the right when kissing.

#### Question 6

No, p-value doesn't prove any statements definitely since it only indicates a probability of a specific situation happening based on the given sample data under the assumption of the null hypothesis. A low p-value indicates the evidence for Fido to be innocent is inconsistent, but does not prove Fido is guilty; in contrast, a high p-value indicates the evidence is inconsistent to prove that Fido is guilty, but does not prove Fido is definitely innocent.

https://chatgpt.com/share/66feec4b-3040-800c-a77e-2f8ed65de41c

The session covered the limitations of using p-values in hypothesis testing. It was explained that a p-value cannot definitively prove whether a null hypothesis is true or false. Instead, a smaller p-value provides stronger evidence against the null hypothesis, but it remains probabilistic. In the context of Fido's innocence or guilt, a low p-value suggests evidence against innocence but doesn't definitively prove guilt, and a high p-value suggests weak evidence for guilt but doesn't definitively prove innocence. Hypothesis testing involves probabilistic reasoning rather than absolute certainty.

#### Question 7

In [5]:
import pandas as pd

patient_data = pd.DataFrame({
    "PatientID": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
    "Age": [45, 34, 29, 52, 37, 41, 33, 48, 26, 39],
    "Gender": ["M", "F", "M", "F", "M", "F", "M", "F", "M", "F"],
    "InitialHealthScore": [84, 78, 83, 81, 81, 80, 79, 85, 76, 83],
    "FinalHealthScore": [86, 86, 80, 86, 84, 86, 86, 82, 83, 84]
})
patient_data

PatientID  Age Gender  InitialHealthScore  FinalHealthScore
0          1   45      M                  84                86
1          2   34      F                  78                86
2          3   29      M                  83                80
3          4   52      F                  81                86
4          5   37      M                  81                84
5          6   41      F                  80                86
6          7   33      M                  79                86
7          8   48      F                  85                82
8          9   26      M                  76                83
9         10   39      F                  83                84

In [6]:
# First let's format this data in the manner of last week's HW "Prelecture" video
# from IPython.display import YouTubeVideo
# YouTubeVideo('Xz0x-8-cgaQ', width=800, height=500)  # https://www.youtube.com/watch?v=Xz0x-8-cgaQ

patient_data['HealthScoreChange'] = patient_data.FinalHealthScore-patient_data.InitialHealthScore
# why do we do the subtraction in this order?
patient_data

PatientID  Age Gender  InitialHealthScore  FinalHealthScore  \
0          1   45      M                  84                86   
1          2   34      F                  78                86   
2          3   29      M                  83                80   
3          4   52      F                  81                86   
4          5   37      M                  81                84   
5          6   41      F                  80                86   
6          7   33      M                  79                86   
7          8   48      F                  85                82   
8          9   26      M                  76                83   
9         10   39      F                  83                84   

   HealthScoreChange  
0                  2  
1                  8  
2                 -3  
3                  5  
4                  3  
5                  6  
6                  7  
7                 -3  
8                  7  
9                  1

In [9]:
print(pd.DataFrame({'HealthScoreChange': patient_data['HealthScoreChange'],
                    '> 0 ?': patient_data['HealthScoreChange']>0}))

random_difference_sign = np.random.choice([-1, 1], size=len(patient_data))
pd.DataFrame({'HealthScoreChange': random_difference_sign*patient_data['HealthScoreChange'].abs(),
              '> 0 ?': (random_difference_sign*patient_data['HealthScoreChange'])>0})

   HealthScoreChange  > 0 ?
0                  2   True
1                  8   True
2                 -3  False
3                  5   True
4                  3   True
5                  6   True
6                  7   True
7                 -3  False
8                  7   True
9                  1   True


HealthScoreChange  > 0 ?
0                 -2  False
1                  8   True
2                  3  False
3                 -5  False
4                 -3  False
5                  6   True
6                 -7  False
7                 -3   True
8                  7   True
9                  1   True

In [10]:
# And then can you see what's happening here???

np.random.seed(1)  # make simulation reproducible
number_of_simulations = 10000  # experiment with this... what does this do?
n_size = len(patient_data)  # 10
IncreaseProportionSimulations_underH0random = np.zeros(number_of_simulations)

# generate "random improvement" proportions assuming H0 (vaccine has no average effect) is true 
# meaning that the "before and after" differences are positive or negative at "random"
for i in range(number_of_simulations):
    
    # why is this equivalent to the suggested idea above?
    random_improvement = np.random.choice([0,1], size=len(patient_data), replace=True)  # <<< `replace=True` ^^^

    # why is .mean() a proportion? 
    IncreaseProportionSimulations_underH0random[i] = random_improvement.mean()
    # why is this the statistic we're interested in? Hint: next section...

In [12]:
# Population parameter under H0
population_parameter_value_under_H0 = 0.5

# Observed test statistic: Proportion of patients with positive health score change
observed_test_statistic = (patient_data.HealthScoreChange > 0).mean()

# Simulated test statistics under H0
simulated_test_statistics = IncreaseProportionSimulations_underH0random

# One-sided: Check if the simulated test statistics are greater than or equal to the observed test statistic
SimTestStats_as_or_more_extreme_than_ObsTestStat = \
    (simulated_test_statistics >= observed_test_statistic)

# Calculate the p-value for the one-sided test
p_value = SimTestStats_as_or_more_extreme_than_ObsTestStat.sum() / number_of_simulations

# Print the results
print("Number of Simulations: ", number_of_simulations, "\n\n",
      "Number of simulated test statistics (under H0)\n",
      'that are "as or more extreme" than the observed test statistic: ',
      SimTestStats_as_or_more_extreme_than_ObsTestStat.sum(), "\n\n",
      'p-value\n(= simulations "as or more extreme" / total simulations): ', p_value, sep="")


Number of Simulations: 10000

Number of simulated test statistics (under H0)
that are "as or more extreme" than the observed test statistic: 565

p-value
(= simulations "as or more extreme" / total simulations): 0.0565


The key change is made one the line \
SimTestStats_as_or_more_extreme_than_ObsTestStat = (simulated_test_statistics >= observed_test_statistic) \
since it makes the program only counts statistics greater than or equal to the observed statistic. The new p=value is calculated as the proportion of simulations where the statistic is greater or equal to the observed statistic. We should generally expect the p-value in a single sided test is smaller than the double sided test under the same condition, since the simulation would have more chance to either be greater or smaller than the observed statistic.

https://chatgpt.com/share/670dfaee-8928-800c-ace7-7669b805e4ca
    
This session focused on refining the hypothesis testing approach, transitioning from a two-sided to a one-sided test. Initially, a two-sided hypothesis test was implemented, comparing the observed test statistic (the proportion of patients who showed improvement in health scores) against the null hypothesis (where the vaccine has no effect and improvements are random). The comparison used the absolute difference between the observed and simulated statistics relative to the null hypothesis, accounting for extremity in both directions.

Next, the code was adjusted to perform a one-sided hypothesis test. This test only considers whether the observed statistic is greater than the hypothesized value, focusing on one direction of extremity. This change leads to a smaller p-value, as it only considers deviations in one direction, unlike the two-sided test, which looks at both.

The discussion clarified that a one-tailed test typically results in a smaller p-value compared to a two-tailed test, making it easier to reject the null hypothesis in the one-sided case, but should only be used when the direction of the effect is clearly hypothesized.

#### Question 8

The purpose of both the Fisher's experiment and this analysis are both to determine whether humans are able to determine the oder of pouring of tea. This two experiment both gives a binary outcome(YES or NO), the null hypothesis is both set to 0.5(guessing), and tested using the comparison between the null hypothesis and the observed statistic. Our sample size is much greater than Fisher's, so we use a z-test instead of the original permulation test, which is a key difference. 

The population is all students in STA130, and the samples are 80 of them. In this specific experiment, the parameter of interest should be p=0.5, which means that the null hypothesis is all students are guessing the answer. In contrast, the null hypothesis is p>0.5, meaning the students have some ability to identify the poruing order of the tea. In this observed sample, 49 of 80 students give the correct answer about the tea, which tells use our sample portion is 0.6125, which is used to compare against the hypothesized proportion of 0.5 under the null hypothesis.

The purpose of this analysis is to determine whether STA130 students are able to taste the order of pouring of the tea or not, in order to address this problem, the statistical hypothesis test is applied to compare the assumption and the observed statistic. First, a one-sample-z-test for a propotion is performed, which include the steps of assume null hypothesis(p=0.5), calculated the test statistic(49/80 = 0.6125) and determine the p-value. A conclusion can be given after we analyze the p-value calculated.

In [13]:
import math
from scipy.stats import norm

# Given values
n = 80  # sample size
x = 49  # number of students who correctly identified the pouring order
p_0 = 0.5  # null hypothesis proportion

# Calculate the sample proportion
p_hat = x / n

# Calculate the standard error under the null hypothesis
standard_error = math.sqrt((p_0 * (1 - p_0)) / n)

# Calculate the z-score for the one-sample proportion test
z_score = (p_hat - p_0) / standard_error

# Calculate the p-value for a one-tailed test (since H1: p > 0.5)
p_value = 1 - norm.cdf(z_score)

# Significance level (commonly set to 0.05)
alpha = 0.05

# Print the results
print(f"Sample proportion (p̂): {p_hat}")
print(f"Z-score: {z_score}")
print(f"P-value: {p_value}")

# Decision
if p_value < alpha:
    print(f"Reject the null hypothesis (p-value = {p_value} < {alpha}). The students are better than random guessing.")
else:
    print(f"Fail to reject the null hypothesis (p-value = {p_value} ≥ {alpha}). No evidence to suggest students are better than guessing.")


Sample proportion (p̂): 0.6125
Z-score: 2.0124611797498115
P-value: 0.022085672454221217
Reject the null hypothesis (p-value = 0.022085672454221217 < 0.05). The students are better than random guessing.


After performing the z-test, we get the p-value of 0.0221, and this value is much smaller than 0.05, which means the null hypothesis has a small proportion to happen under the observed data. This p-value suggests a strong evidence for students are able to determine the order of pouring. Therefore, we failed to claim the null hypothesis is true since there's no enough evidence. 

https://chatgpt.com/share/670e02be-1e94-800c-9ee3-8d4bf74c1fdb

In this session, an analysis was conducted to evaluate whether STA130 students could correctly identify whether milk or tea was poured first in a cup of tea. The null hypothesis stated that the students were guessing, with a 50% chance of being correct, while the alternative hypothesis suggested that students were better than random guessing.

A one-sample z-test for a proportion was used, with the observed sample proportion being 0.6125 (49 out of 80 students were correct). The z-score was calculated as 2.01, leading to a p-value of approximately 0.0221. Since this p-value is less than the significance level of 0.05, the null hypothesis was rejected, indicating that the students were statistically better than random guessing.

This experiment shares key similarities with Fisher's tea-tasting experiment involving Dr. Muriel Bristol, both testing whether observed outcomes deviate from random guessing. The primary differences lie in sample size, the type of test used, and the experimental design.

#### Question 9

Yes.

#### Chatbot history

https://chatgpt.com/share/66fed021-434c-800c-bf4d-98e4d3db77c6
Statistical examination requires ideas to be measurable, meaning they must involve quantifiable variables that can be objectively collected and analyzed. Without clear metrics, it becomes difficult to apply statistical methods.

https://chatgpt.com/share/66fed6e9-0fd8-800c-ab16-5a227177c665
This session focused on clarifying key statistical concepts related to hypothesis testing, specifically the distinction between observed sample values (( x_i )), the sample mean (( \bar{x} )), the population mean (( \mu )), and the hypothesized population mean under the null hypothesis (( \mu_0 )).
Observed sample values (( x_i )) are individual data points collected from a sample.
The sample mean (( \bar{x} )) is the average of those values and serves as an estimate of the population mean, though it varies between samples.
The population mean (( \mu )) is a fixed but unknown value that represents the true average of the population.
The hypothesized mean (( \mu_0 )) is the value assumed under the null hypothesis, representing a baseline for comparison in the hypothesis test.
The distinction between these concepts highlights the goal of hypothesis testing: using sample data to draw conclusions about the population mean.

https://chatgpt.com/share/66fedad8-2fc0-800c-99a7-b23227c2978b
In this session, two key concepts in hypothesis testing were covered:
Imagining a World Where the Null Hypothesis is True: When calculating a p-value, the assumption is made that the null hypothesis is true to evaluate how likely the observed data would be under this assumption. This helps in deciding whether to reject or fail to reject the null hypothesis based on the extremity of the data.
What a p-value is: A p-value represents the probability of obtaining a result as extreme, or more extreme, than the observed data, assuming the null hypothesis is true. A small p-value suggests evidence against the null hypothesis, while a large p-value indicates that the data is not unusual under the null hypothesis. It does not measure the probability that the null hypothesis is true or false, nor does it quantify the size or importance of an effect.

https://chatgpt.com/share/66fedad8-2fc0-800c-99a7-b23227c2978b
The observed test statistic is compared to the sampling distribution of the test statistic under the assumption that the null hypothesis is true. If the observed value falls near the center of the distribution, it suggests that the data aligns with the null hypothesis. However, if the observed statistic falls in the tails (indicating an extreme value), it suggests the result is unusual under the null hypothesis. This comparison helps determine the p-value, with more extreme test statistics leading to smaller p-values and stronger evidence against the null hypothesis.

https://chatgpt.com/share/66fee8a9-9020-800c-94aa-c3f741d7e98a
In this session, a simulation was conducted to estimate a p-value for a study on head-tilting preferences among kissing couples. The null hypothesis assumed no preference for tilting direction, with a 50/50 chance of tilting left or right. The observed data showed that 80 out of 124 couples tilted their heads to the right, a proportion of 64.5%.
A binomial model representing the 50/50 probability assumption was used, and 10,000 trials were run. The resulting p-value was 0.001, indicating that the likelihood of observing 80 or more right tilts, under the assumption of no preference, is just 0.1%.
This low p-value suggests strong evidence against the null hypothesis, implying a possible preference for tilting to the right when kissing.

https://chatgpt.com/share/66feec4b-3040-800c-a77e-2f8ed65de41c
The session covered the limitations of using p-values in hypothesis testing. It was explained that a p-value cannot definitively prove whether a null hypothesis is true or false. Instead, a smaller p-value provides stronger evidence against the null hypothesis, but it remains probabilistic. In the context of Fido's innocence or guilt, a low p-value suggests evidence against innocence but doesn't definitively prove guilt, and a high p-value suggests weak evidence for guilt but doesn't definitively prove innocence. Hypothesis testing involves probabilistic reasoning rather than absolute certainty.

https://chatgpt.com/share/670dfaee-8928-800c-ace7-7669b805e4ca
This session focused on refining the hypothesis testing approach, transitioning from a two-sided to a one-sided test. Initially, a two-sided hypothesis test was implemented, comparing the observed test statistic (the proportion of patients who showed improvement in health scores) against the null hypothesis (where the vaccine has no effect and improvements are random). The comparison used the absolute difference between the observed and simulated statistics relative to the null hypothesis, accounting for extremity in both directions.
Next, the code was adjusted to perform a one-sided hypothesis test. This test only considers whether the observed statistic is greater than the hypothesized value, focusing on one direction of extremity. This change leads to a smaller p-value, as it only considers deviations in one direction, unlike the two-sided test, which looks at both.
The discussion clarified that a one-tailed test typically results in a smaller p-value compared to a two-tailed test, making it easier to reject the null hypothesis in the one-sided case, but should only be used when the direction of the effect is clearly hypothesized.

https://chatgpt.com/share/670e02be-1e94-800c-9ee3-8d4bf74c1fdb
In this session, an analysis was conducted to evaluate whether STA130 students could correctly identify whether milk or tea was poured first in a cup of tea. The null hypothesis stated that the students were guessing, with a 50% chance of being correct, while the alternative hypothesis suggested that students were better than random guessing.
A one-sample z-test for a proportion was used, with the observed sample proportion being 0.6125 (49 out of 80 students were correct). The z-score was calculated as 2.01, leading to a p-value of approximately 0.0221. Since this p-value is less than the significance level of 0.05, the null hypothesis was rejected, indicating that the students were statistically better than random guessing.
This experiment shares key similarities with Fisher's tea-tasting experiment involving Dr. Muriel Bristol, both testing whether observed outcomes deviate from random guessing. The primary differences lie in sample size, the type of test used, and the experimental design.